### ISSUE TITLE
Disable hadolint DL3008

### ISSUE THREAD

**Issue by dlukeomalley** [returntocorp/bento#202]
## Overview
Hadolint DL3008 fires often and in practice it seems like few people rigorously pin their `apt-get` package versions. For the users who I've interviewed that have seen this, they felt like this was a best practice, but not something they'd do in practice (too onerous).

This ticket proposes disabling DL3008

```shell
warning: r2c.hadolint.DL3008
     --> Dockerfile:2:1
      |
    2 |   RUN apt-get update && apt-get -y install python-pip python-dev npm git libpq-dev curl unzip
      |
      = note: Pin versions in apt get install. Instead of `apt-get install <package>` use `apt-get install <package>=<version>`
```

## Current Behavior
> Describe the actual behavior, and provide the content of `~/.bento/last.log` if possible. Screenshots and text dumps are particularly helpful.

## Steps to Reproduce

1. `git clone https://github.com/opensourceactivismtech/call-power/`
2. `bento init`
3. `bento enable tool r2c.hadolint`
4. `bento check`

## Possible Solution
Disable DL3008

## Additional Context
Bento version (run `bento --version`): 0.7.0b2

**Comment by minusworld** [comment]
@dlukeomalley Beginning to look at this now. Do we want to disable version pinning for other package mangers too? DL3013 is pip, DL3016 is npm, and DL3018 is apk

**Comment by dmontagu** [comment]
My 2c, for what it's worth:

Regarding `pip`, `npm`, and other non-system package managers (like `apt-get` or `apk`):
* I think any dependency installed via `pip`, `npm`, etc. should at the very least have its version *recorded* somewhere for the sake of rollback (pinning obviously offers additional benefits).
* At the same time, I *don't* think pinning should be done inside a dockerfile, but instead inside an easily-generated/updated lockfile specific to the relevant package manager, which is used during the docker build. (A multi-stage build can be used to remove build system cruft if necessary.)
* So I personally think it makes sense to leave the checks on for `pip and `npm`, if only to discourage the use of raw `pip install` or `npm install` lines in a dockerfile. Admittedly there are cases where it makes sense to use unpinned pip installs (typically in lightweight, self-contained example dockerfiles), but I personally would prefer that it still require manually marking it as ignored, if only to ensure this consideration isn't missed by new dockerfile-writing developers.

Regarding `apt-get`:
* I wish there was a highly convenient lockfile process (similar to what `pipenv`/`poetry` and `npm`/`yarn` are capable of) for system dependencies, but I'm not aware of one.
    * (If a tool for this *does* exist that works well in most cases and is convenient to use/scriptable, I would strongly encourage leaving the relevant checks on, and also publicizing the tool!)
* Given the relative stability of the `apt-get` package ecosystem, combined with the lack of a convenient tool for managing these dependencies, I personally think the costs of pinning outweigh the benefits

The main difference to me is one of convenience -- while it requires a little additional work to get `npm`/`pip` using pinned dependencies, the benefits are potentially large and the on-going cost is low; the reverse generally seems to be the case for `apt-get`/`apk`. (I'd be happy to be contradicted though!)

**Comment by minusworld** [comment]
Awesome @dmontagu! Thanks for the thoughtful response!

**Comment by dlukeomalley** [comment]
Fixed in 0.8 🎉
